pkg/util/zookeeper_util.go

```diff
@@ -0,0 +1,120 @@
+/**
+ * Copyright (c) 2018 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package util
+
+import (
+	"container/list"
+	"context"
+	"fmt"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/pravega/pravega-operator/pkg/apis/pravega/v1alpha1"
+	"github.com/samuel/go-zookeeper/zk"
+)
+
+const (
+	// Set in https://github.com/pravega/pravega/blob/master/docker/bookkeeper/entrypoint.sh#L21
+	PravegaPath = "pravega"
+	ZkFinalizer = "cleanUpZookeeper"
+)
+
+// Wait for pods in cluster to be terminated
+func WaitForClusterToTerminate(kubeClient client.Client, p *v1alpha1.PravegaCluster) (err error) {
```

Comment:
This function fits better in `k8sutil.go` as it does not contain any ZK logic.

Reply:
Got it